frontend/src/metabase/parameters/components/ParameterValuePicker/OwnDatePicker.tsx

```diff
@@ -0,0 +1,153 @@
+import { useClickOutside } from "@mantine/hooks";
+import { useState } from "react";
+import { t } from "ttag";
+
+import { DateAllOptionsWidget } from "metabase/components/DateAllOptionsWidget";
+import { DateMonthYearWidget } from "metabase/components/DateMonthYearWidget";
+import { DateQuarterYearWidget } from "metabase/components/DateQuarterYearWidget";
+import { DateRelativeWidget } from "metabase/components/DateRelativeWidget";
+import { formatParameterValue } from "metabase/parameters/utils/formatting";
+import { Popover } from "metabase/ui";
+import type {
+  DateParameterType,
+  Parameter,
+  ParameterType,
+} from "metabase-types/api";
+
+import {
+  TextInputIcon,
+  TextInputTrirgger,
+} from "./ParameterValuePicker.styled";
+
+export function OwnDatePicker(props: {
```

Comment:
nit: In some newer files (but not all) we put nested components in subfolders (take a look at `metabase/querying`). We had an RFC about this and it was accepted, but I don't think that all people follow this approach